doc/api/esm.md

```diff
@@ -239,11 +241,35 @@ const { default: barData } =
   await import('./bar.json', { assert: { type: 'json' } });
 ```
 
-Node.js supports the following `type` values:
+Node.js supports the following `type` values, for which the assertion is
+mandatory:
```

Comment:
I got whiplash following whether this will always be mandatory. If they won't be, is that worth mentioning/hinting (maybe just dropping in a "currently")?

Reply:
Any non-`'javascript'` types are always mandatory. Per the HTML spec, which we’re following here, if there’s no `type` specified then the `type` is `'javascript'`.

See https://github.com/nodejs/node/discussions/40790#discussioncomment-1660814 (and maybe move the discussion to there) for the question whether Wasm could/should be considered `'javascript'`, and therefore not need an explicit type, or if it should be considered some other type which would then be mandatory. In the PR as written currently, Wasm maps to `type: 'webassembly'`, and since it’s not `'javascript'`, the `type: 'webassembly'` is mandatory.

Reply:
ugh, sorry meant to resolve the whole thread, but I think I can't and inadvertently just hide my own comment 🤦‍♂️